qiskit_experiments/data_processing/nodes.py

```diff
@@ -411,6 +433,135 @@ def _process(self, data: np.array) -> np.array:
         return unp.sqrt(data[..., 0] ** 2 + data[..., 1] ** 2) * self.scale
 
 
+class Discriminator(IQPart):
+    """A class to discriminate level 1 data, e.g., IQ data, to produce counts.
+
+    This node can be seen as a wrapper for a discriminator object such as a Sklearn classifier.
+    The wrapped discriminator object must have a method :meth:`predict` that takes as input
+    a list of lists and returns a list of labels as do most Sklearn classifiers. Crucially,
+    this node can be initialized with a single discriminator which applies to each memory
+    slot or it can be initialized with a list of discriminators, i.e. one for each slot.
+    """
+
+    def __init__(
+        self, discriminator: Union[Any, List[Any]], validate: bool = True, no_ufloat: bool = True
+    ):
+        """Initialize the node with an object that can discriminate.
+
+        Args:
+            discriminator: The entity that will perform the discrimination. This needs to
+                be an object or a list of objects with a :meth:`predict` method that takes
+                as input a list of lists and returns a list of labels. If a list of
+                discriminators is given then there should be as many discriminators as there
+                will be slots in the memory.
+            no_ufloat: A boolean which if set to True, i.e., the default, will cause the node
+                to convert the data from uncertainties to regular floats.
+        """
+        super().__init__(1.0, validate)
+        self._discriminator = discriminator
+        self._no_ufloat = no_ufloat
+
+    def _format_data(self, data: np.ndarray) -> np.ndarray:
+        """Check that there are as many discriminators as there are slots."""
+        data = super()._format_data(data)
+
+        if self._validate:
+            if isinstance(self._discriminator, list):
+                if self._n_slots != len(self._discriminator):
+                    raise DataProcessorError(
+                        f"The Discriminator node has {len(self._discriminator)} which does "
+                        f"not match the {self._n_slots} slots in the data."
+                    )
+
+        if self._no_ufloat:
```

Comment:
I'm fine with dropping ufloat entirely here de-facto for now. This would mean removing the `no_ufloat` option. One could think of perhaps confidence intervals on the discrimination based on how close a point is to the boundary etc but that seems out of scope for this PR and would need some careful thinking.

Reply:
I removed `no_ufloat` here: [2834d2e](https://github.com/Qiskit/qiskit-experiments/pull/805/commits/2834d2e82847eac84efa45322a29c5298b5f7763)